rfcs/2020-08-27-3603-postgres-metrics.md

```diff
@@ -0,0 +1,154 @@
+# RFC 3603 - 2020-08-27 - Collecting metrics from PostgreSQL
+
+This RFC is to introduce a new metrics source to consume metrics from PostgreSQL database servers. The high level plan is to implement one source that collects metrics from PostgreSQL server instances.
+
+Background reading on PostgreSQL monitoring:
+
+- https://www.datadoghq.com/blog/postgresql-monitoring/
+
+## Scope
+
+This RFC will cover:
+
+- A new source for PostgreSQL server metrics.
+
+This RFC will not cover:
+
+- Other databases.
+
+## Motivation
+
+Users want to collect, transform, and forward metrics to better observe how their PostgreSQL databases are performing.
+
+## Internal Proposal
+
+Build a single source called `postgresql_metrics` (name to be confirmed) to collect PostgreSQL metrics.
+
+The recommended implementation is to use the Rust PostgreSQL client to connect the target database server by address specified in configuration.
+
+- https://docs.rs/postgres/0.17.5/postgres/index.html
+
+The source would then run the following queries:
+
+- `SELECT * FROM pg_stat_database`
+- `SELECT * FROM pg_stat_database_conflicts`
+- `SELECT * FROM pg_stat_bgwriter`
+
+And return these metrics by parsing the query results and converting them into metrics using the database name and column names.
+
+- `postgresql_up` -> Used as an uptime metric (0/1) ? - merits a broader discussion.
```

Comment:
These "is the service up" metrics have been useful to me in the past. I suggest we have one for each metrics source.

Reply:
+1.